Library/Homebrew/cmd/uninstall.rb

```diff
@@ -9,15 +10,39 @@ def uninstall
       ARGV.kegs.each do |keg|
         keg.lock do
           puts "Uninstalling #{keg}... (#{keg.abv})"
+
+          # Find symlinks that can point to keg.rack
+          links = HOMEBREW_CELLAR.subdirs.select { |d| d.symlink? }
```

Comment:
We can move it outside of `ARGV.kegs.each` so you don't need to check it repeatedly and can avoid the repeat codes in below. Also you can just do `.select(&:symlink?)` instead of block.